internal/cli/atlas/dbusers/create.go

```diff
@@ -26,17 +26,41 @@ import (
 	"github.com/mongodb/mongocli/internal/output"
 	"github.com/mongodb/mongocli/internal/store"
 	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/mongodb/mongocli/internal/validate"
 	"github.com/spf13/cobra"
 	atlas "go.mongodb.org/atlas/mongodbatlas"
 )
 
 type CreateOpts struct {
 	cli.GlobalOpts
-	username string
-	password string
-	authDB   string
-	roles    []string
-	store    store.DatabaseUserCreator
+	username   string
+	password   string
+	x509Type   string
+	awsIamType string
+	roles      []string
+	store      store.DatabaseUserCreator
+}
+
+const (
+	AWSIAMTypeUser   = "USER"
+	AWSIAMTypeRole   = "ROLE"
+	AWSIAMTypeNone   = "NONE"
+	X509TypeManaged  = "MANAGED"
+	X509TypeCustomer = "CUSTOMER"
+	X509TypeNone     = "NONE"
+)
+
+var (
+	validX509Flags   = []string{X509TypeNone, X509TypeManaged, X509TypeCustomer}
+	validAWSIAMFlags = []string{AWSIAMTypeNone, AWSIAMTypeRole, AWSIAMTypeUser}
+)
+
+func (opts *CreateOpts) isX509Set() bool {
+	return opts.x509Type == X509TypeCustomer || opts.x509Type == X509TypeManaged
+}
+
+func (opts *CreateOpts) isAWSIAMSet() bool {
+	return opts.awsIamType == AWSIAMTypeUser || opts.awsIamType == AWSIAMTypeRole
```

Comment:
similar comment from the x509 PR, it's easier to read just to say 
```suggestion
	return opts.awsIamType != AWSIAMTypeNone
```